contracts/libraries/FVMLib.sol

```diff
@@ -167,6 +168,9 @@ function encodeJumpInstruction(bytes[] memory instructions)
     returns (bytes memory)
 {
     uint8 totalInstructions = uint8(instructions.length);
+
+    if (totalInstructions > 255) revert TooManyInstructions();
```

Comment:
Since `totalInstructions` has already been cast to a `uint8`, its maximum value is 255 so the revert is unreachable. Perhaps you meant to check `instructions.length` directly?

Reply:
I think the value `255` was used as a placeholder since there might be a lower limit related to the gas limit? Still unsure about this one.

Reply:
Well, this fails to address the concern from the finding, which was that someone might try to encode, say, a 256 instruction sequence, and this function will silently truncate the encoding at 255 instructions, potentially leading to unexpected outcomes if the caller doesn't notice. The check needs to be done on `instructions.length` to catch that.

Reply:
Yep just moving up the check to be instructions.length (a uint256) before casting fixes it

Reply:
yes, the check needs to be done on the value before you truncate it to the uint8 range with `uint8(instructions.length);`.

Reply:
Fixed now!

Reply:
fixed